purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -4190,18 +4185,19 @@ class PurchasesTest {
         type: ProductType,
         offeringIdentifier: String?
     ): ReceiptInfo {
-        val skuDetails = stubSkuDetails(
-            productId = sku,
-            price = 2.00,
-            subscriptionPeriod = if (type == ProductType.SUBS) "P1M" else "",
-            introductoryPricePeriod = if (type == ProductType.SUBS) "P7D" else null,
-            freeTrialPeriod = if (type == ProductType.SUBS) "P7D" else null
-        )
+        val productDetails =
+            if (type == ProductType.SUBS) createMockProductDetailsFreeTrial(sku, 2.00)
+            else createMockOneTimeProductDetails(sku, 2.00)
+
+        val storeProduct = if (type == ProductType.SUBS) productDetails.toStoreProduct(
+            productDetails.subscriptionOfferDetails!![1],
```

Comment:
why is this one grabbing `[1]`?

Reply:
Ah yeah, need to clean this up. Basically the base plan offer is the second one in the mocked `ProductDetails`. However, we can compute this so its more robust. Good catch!

Reply:
Oh I see, it's grabbing the hardcoded base plan offer